src/uu/sort/src/sort.rs

```diff
@@ -342,30 +342,20 @@ impl GlobalSettings {
     fn parse_byte_count(input: &str) -> Result<usize, ParseSizeError> {
         // GNU sort (8.32)   valid: 1b,        k, K, m, M, g, G, t, T, P, E, Z, Y
         // GNU sort (8.32) invalid:  b, B, 1B,                         p, e, z, y
-        const ALLOW_LIST: &[char] = &[
-            'b', 'k', 'K', 'm', 'M', 'g', 'G', 't', 'T', 'P', 'E', 'Z', 'Y',
-        ];
-        let mut size_string = input.trim().to_string();
-
-        if size_string.ends_with(|c: char| ALLOW_LIST.contains(&c) || c.is_ascii_digit()) {
-            // b 1, K 1024 (default)
-            if size_string.ends_with(|c: char| c.is_ascii_digit()) {
-                size_string.push('K');
-            } else if size_string.ends_with('b') {
-                size_string.pop();
-            }
-            let size = parse_size(&size_string)?;
-            usize::try_from(size).map_err(|_| {
-                ParseSizeError::SizeTooBig(format!(
-                    "Buffer size {} does not fit in address space",
-                    size
-                ))
-            })
-        } else if size_string.starts_with(|c: char| c.is_ascii_digit()) {
-            Err(ParseSizeError::InvalidSuffix("invalid suffix".to_string()))
-        } else {
-            Err(ParseSizeError::ParseFailure("parse failure".to_string()))
-        }
+        let size = Parser::default()
+            .with_allow_list(&[
+                "b", "k", "K", "m", "M", "g", "G", "t", "T", "P", "E", "Z", "Y",
+            ])
+            .with_default_unit("K")
+            .with_b_byte_count(true)
+            .parse(input.trim())?;
+
+        usize::try_from(size).map_err(|_| {
+            ParseSizeError::SizeTooBig(format!(
+                "Buffer size {} does not fit in address space",
```

Comment:
can you add a test which reproduces this error ?

Reply:
ok, I added it.

Reply:
What you added is not really a valid test, because you just tested Rust's `try_from` separately from the code (and it should have been map_err instead of err). @sylvestre it's not possible to hit that code path in 64 bit systems, because `u64` to `usize` conversion should never fail there.

Reply:
my bad, I fixed `map` -> `map_err`.

Reply:
Still not really a meaningful test, but like I said, that's not really your fault, because there's no good way to test this path on a 64-bit machine. So I think we can remove it.